src/acuvity/guard/config.py

```diff
@@ -0,0 +1,279 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+import yaml
+
+from .constants import GuardName
+from .errors import ConfigValidationError, GuardConfigError, ThresholdParsingError
+from .threshold import Threshold
+
+
+@dataclass(frozen=True)
+class Match:
+    """Immutable match configuration"""
+    threshold: Threshold
+    redact: bool = False
+    count_threshold: int = 0
+
+@dataclass(frozen=True)
+class Guard:
+    """Immutable guard configuration"""
+    name: GuardName
+    matches: Dict[str, Match]
+    threshold: Threshold
+    count_threshold: int = 0
+
+class GuardConfig:
+    """
+    Parser for guard configuration files.
+
+    This class handles parsing of guard configuration files in YAML format,
+    validating their contents, and converting between analyzer names and IDs.
+
+    The parser handles two types of guards:
+    1. Match Guards: Guards with a 'matches' section (e.g., pii_detector)
+    2. Simple Guards: Guards without matches (e.g., prompt_injection, toxicity)
+    """
+
+    DEFAULT_THRESHOLD = Threshold("> 0.0")
+    def __init__(self, config: Optional[Union[str, Path, Dict, List[Guard]]] = None):
+        """
+        Initialize parser with analyzer mapping.
+
+        Args:
+            config: Configuration as a string, filepath or dictionary
+        """
+        self._parsed_guards: List[Guard] = []
+
+        # Handle default configuration
+        if config is None:
+            for guard in GuardName:
+                # skip keyword detector in default
+                if guard != GuardName.KEYWORD_DETECTOR:
+                    self._parsed_guards.append(Guard(
+                        name=guard,
+                        matches={},
+                        threshold=self.DEFAULT_THRESHOLD,
+                        count_threshold=0,
+                    ))
+            return
+
+        # Use the config provided
+        self._parse_config(config)
+
+    @staticmethod
+    def load_yaml(path: Union[str, Path]) -> Dict[str, Any]:
+        """
+        Load and parse YAML file.
+
+        Args:
+            path: Path to YAML file
+
+        Returns:
+            Parsed YAML content as dictionary
+
+        Raises:
+            GuardConfigError: If file cannot be read or parsed
+        """
+        try:
+            with open(path, encoding='utf-8') as f:
+                return yaml.safe_load(f)
+        except (yaml.YAMLError, OSError) as e:
+            raise GuardConfigError(f"Failed to load config file: {e}") from e
+
+    def _parse_config(self, config: Union[str, Path, Dict, List[Guard]]) -> List[Guard]:
+        """
+        Parse guard configuration from file or dictionary.
+
+        Args:
+            config: Path to YAML file or dictionary containing configuration
```

Comment:
config: path to YAML file, dictionary containing guard configuration, or a direct list of typed guards